### ISSUE TITLE
`metadata.device.description` not always sent in a webhook

### ISSUE THREAD

**Issue by jobannon** [FusionAuth/fusionauth-issues#2100]
### Description 
This may be working as designed.

Metadata.device.description is not always sent over.  This is an optional parameter, so it could be not available in certain circumstances.

### Affects versions
List the version you are running, and any other versions you have attempted a recreated. 

### Steps to reproduce
1. On the authorize page add a field such as:

```
<input type="hidden" name="metaData.device.description" value="cid:abc123"/>
```

2. Register a webhook for the `user.login.success`
3. Login
3. See that we are sending a `cid:abc123` value in the event
4. Open an adjacent tab.
5. Login via SSO session.
6. See that the metadata is not transmitted in the webhook event.

### Expected behavior
This current state is expected, we should document why.  If this is a bug, we should fix it.  I would expect that each time we hit the authorization endpoint if we send metadata, it should be transmitted.

### Platform
(Please complete the following information)
 - Device: [e.g. Desktop, iPhone X, Pixel 3, etc]
 - OS: [e.g. iOS, macOS, Windows ME]
 - Browser + version [e.g. chrome, safari plus a version]
 - Database [MySQL | PostgresSQL] & and version

### Community guidelines
All issues filed in this repository must abide by the [FusionAuth community guidelines](https://fusionauth.io/community/forum/topic/1000/code-of-conduct).

**Comment by mooreds** [comment]
This is a bit of a hack and we should probably solve the root issue via https://github.com/FusionAuth/fusionauth-issues/issues/1660 imo.

**Comment by robotdan** [comment]
I don't think this is a hack. It may be a hack if you are using the device meta data to pass along information un-related to the device. 

But the event info that is sent in an event is supposed to contain device information. So if the device information is not being sent correctly when the user is already logged into SSO, there may be a bug. 

Or - it could be that we are intentionally not updating the device description on an SSO login, TBD.

**Comment by asasd325** [comment]
I'm happy to see the issue is now solved. Thank you for updating us with the outcome.

**Comment by robotdan** [comment]
@jobannon can we close this issue?

If we do need to keep it open, can you update the title and description to be more specific? For example, what does "not always" mean?
 
 > Metadata.device.description is not always sent over. This is an optional parameter, so it could be not available in certain circumstances.

What circumstances is it not sent?

**Comment by robotdan** [comment]
Internal:
See if we can recreate, and if not, close.

**Comment by bhalsey** [comment]
The reproduction steps have the behavior indicated in this issue of not passing along the metadata.device.description to the webhook.  But this is because the added hidden input field is never used in an SSO login.  The GET call to `/oauth2/authorize` doesn't render the login form and instead does a redirect back to the customer application with a code. 

An additional workaround of adding a url query parameter of `metadata.device.description=somevalue` on the GET request to /oauth2/authorize will result in the webhook event having this data.

Confirmed with @robotdan this is working as designed.  The root issue is better addressed in #1660 .

**Comment by mooreds** [comment]
@bhalsey can you please update the workaroudn in #1660 with the additional requirement (of sending along a query parameter too)?
